### PR TITLE
UF-7414: Comfact Integration

### PR DESCRIPTION
**Pull request by cheezi0747**
Also contains
UF-7415: Service logic
UF-7414: API-design

### CODE REVIEW

File: src/main/java/se/sundsvall/comfactfacade/api/SigningResource.java [hunk 1]
```diff
@@ -40,38 +44,54 @@ public class SigningResource {
 
 	public SigningResource(final SigningService signingService) {this.signingService = signingService;}
 
+	@GetMapping
```
Comment (AndreSoderlund) [inline src/main/java/se/sundsvall/comfactfacade/api/SigningResource.java:47]: produces = {APPLICATION_PROBLEM_JSON_VALUE, APPLICATION_JSON VALUE}

File: src/main/java/se/sundsvall/comfactfacade/api/SigningResource.java [hunk 2]
```diff
@@ -40,38 +44,54 @@ public class SigningResource {
...
+	@Operation(summary = "Get all signing instances.")
+	@ApiResponse(responseCode = "200", description = "Successful operation", useReturnTypeSchema = true)
+	public ResponseEntity<List<SigningInstance>> getSigningRequests() {
+		return ResponseEntity.ok(signingService.getSigningRequests());
+	}
+
+	@GetMapping(path = "{signingId}", produces = {APPLICATION_PROBLEM_JSON_VALUE})
```
Comment (AndreSoderlund) [inline src/main/java/se/sundsvall/comfactfacade/api/SigningResource.java:54]: produces = {APPLICATION_PROBLEM_JSON_VALUE, APPLICATION_JSON VALUE}

File: src/main/java/se/sundsvall/comfactfacade/api/SigningResource.java [hunk 3]
```diff
@@ -40,38 +44,54 @@ public class SigningResource {
...
-		return ResponseEntity.ok(signingService.getStatus(signingId));
+	public ResponseEntity<Void> cancelSigningRequest(@PathVariable final String signingId) {
+		signingService.cancelSigningRequest(signingId);
+		return ResponseEntity.noContent().build();
 	}
 
-
-	@GetMapping(path = "{signingId}/document", produces = {APPLICATION_PROBLEM_JSON_VALUE})
-	@Operation(summary = "Get signed document.")
+	@GetMapping(path = "{signingId}/signatory/{partyId}", produces = {APPLICATION_PROBLEM_JSON_VALUE})
```
Comment (AndreSoderlund) [inline src/main/java/se/sundsvall/comfactfacade/api/SigningResource.java:88]: produces = {APPLICATION_PROBLEM_JSON_VALUE, APPLICATION_JSON VALUE}

File: src/main/java/se/sundsvall/comfactfacade/api/model/Party.java [hunk 4]
```diff
@@ -1,42 +1,89 @@
...
+
 
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@SuperBuilder(setterPrefix = "with")
 @Schema(description = "A party related to the signing process.")
-public record Party(
+public class Party {
```
Comment (AndreSoderlund) [inline src/main/java/se/sundsvall/comfactfacade/api/model/Party.java:25]: Is this class only used as a response? In that case, do we need validation?

Comment (cheezi0747) [reply src/main/java/se/sundsvall/comfactfacade/api/model/Party.java:25]: Its used both in the request and in the response

File: src/test/java/se/sundsvall/comfactfacade/api/SigningResourceTest.java [hunk 5]
```diff
@@ -35,152 +35,145 @@ class SigningResourceTest {
...
 			.exchange()
-			.expectStatus().isNoContent();
+			.expectStatus()
+			.isCreated()
+			.expectHeader()
+			.location("/signings/someSigningId");
 
 		// Assert
-		verify(signingServiceMock).cancelSigningRequest(signingId);
+		verify(signingServiceMock).createSigningRequest(any(SigningRequest.class));
```
Comment (josc04) [inline src/test/java/se/sundsvall/comfactfacade/api/SigningResourceTest.java:89]: Why not use the actual request object to validate that correct data is sent to service (i.e `verify(signingServiceMock).createSigningRequest(signingRequest);`)? 

Comment goes for all verifcations where `any(...class)` is used.

File: src/test/java/se/sundsvall/comfactfacade/api/model/DocumentTest.java [hunk 6]
```diff
@@ -18,10 +35,16 @@ void constructorAndGetter() {
...
-		assertThat(result.name()).isEqualTo(name);
-		assertThat(result.fileName()).isEqualTo(fileName);
-		assertThat(result.mimeType()).isEqualTo(mimeType);
-		assertThat(result.content()).isEqualTo(content);
+		assertThat(result.getName()).isEqualTo(name);
+		assertThat(result.getFileName()).isEqualTo(fileName);
+		assertThat(result.getMimeType()).isEqualTo(mimeType);
+		assertThat(result.getContent()).isEqualTo(content);
+	}
+
```
Comment (josc04) [inline src/test/java/se/sundsvall/comfactfacade/api/model/DocumentTest.java:43]: Test with builder methods?

Comment goes for all api model classes with builder methods where the builder methods are not tested.

File: src/test/java/se/sundsvall/comfactfacade/service/SigningServiceTest.java [hunk 7]
```diff
@@ -1,56 +1,129 @@
...
+	void updateSigningRequest() {
+		// Arrange
+		final var signingId = "someSigningId";
+		final var signingRequest = new SigningRequest();
+
+		// Act
+		signingService.updateSigningRequest(signingId, signingRequest);
+
+		// Assert
+		verify(comfactClient).updateSigningInstance(any());
```
Comment (josc04) [inline src/test/java/se/sundsvall/comfactfacade/service/SigningServiceTest.java:54]: Use captor to verify that correct values are propagated to client mock?

Comment goes for all calls to client.

File: src/main/java/se/sundsvall/comfactfacade/integration/comfact/ComfactIntegration.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+
+	public ComfactIntegration(final ComfactClient comfactClient, final ComfactProperties comfactProperties) {
+		this.comfactClient = comfactClient;
+		this.comfactProperties = comfactProperties;
+	}
+
+	private void handleResponse(final ResponseType response) {
+		if (!response.getResult().isSuccess()) {
+			throw Problem.valueOf(Status.INTERNAL_SERVER_ERROR, response.getResult().getResultMessage());
```
[the remark this answers is not in the document]
Comment (cheezi0747) [reply src/main/java/se/sundsvall/comfactfacade/integration/comfact/ComfactIntegration.java:38]: That would be logical if the underlying service was returning a traditional HTTP error. However the underlying service is returning this as an object in the response.

`response.getResult().isSuccess()` is not checking if the statuscode `.isSuccessful()` its checking a boolean value in the object that was returned.

The response result can also contain one of around 50 of custom Response Result Codes. In theory we could probably create a mapping of which of these result codes corresponds to a 500 or a 502. I however think a generic 500 is good enough for this.

File: src/main/java/se/sundsvall/comfactfacade/api/SigningResource.java [hunk 9]
```diff
@@ -40,38 +45,58 @@ public class SigningResource {
...
 
-	@PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = {APPLICATION_PROBLEM_JSON_VALUE})
-	@Operation(summary = "Create Signing request.")
+	@GetMapping(produces = {APPLICATION_JSON_VALUE, APPLICATION_PROBLEM_JSON_VALUE})
+	@Operation(summary = "Get all signing instances.")
 	@ApiResponse(responseCode = "200", description = "Successful operation", useReturnTypeSchema = true)
-	public ResponseEntity<Void> createSigningRequest(@Valid @RequestBody final SigningRequest signingRequest) {
-		signingService.createSigningRequest(signingRequest);
-		return ResponseEntity.ok().build();
+	public ResponseEntity<List<SigningInstance>> getSigningRequests() {
```
[the remark this answers is not in the document]
Comment (cheezi0747) [reply src/main/java/se/sundsvall/comfactfacade/api/SigningResource.java:51]: Will be handled in a seperate story as pagination lives in a different API document we don't currently have.

**Pull request merged**